src/test/java/org/broadinstitute/hellbender/utils/downsampling/ReadsDownsamplingIteratorUnitTest.java

```diff
@@ -132,6 +133,22 @@ public void testRemoveThrows() {
         iter.remove();
     }
 
+    @Test
+    public void testReadsDownsamplingIteratorWithReservoirDownsampler() {
+        final int TOTAL_READ_COUNT = 100;
+        final int TARGET_COVERAGE = 45;
+        final List<GATKRead> reads = new ArrayList<>(TOTAL_READ_COUNT);
+        IntStream.range(1, TOTAL_READ_COUNT).forEach(i -> reads.add(readWithName(Integer.toString(i))));
+
+        final List<GATKRead> downsampledReads = new ArrayList<>();
+        final ReadsDownsamplingIterator downsamplingIter = new ReadsDownsamplingIterator(reads.iterator(), new ReservoirDownsampler(TARGET_COVERAGE));
+        for ( final GATKRead read : downsamplingIter ) {
+            downsampledReads.add(read);
+        }
+
+        Assert.assertEquals(downsampledReads.size(), TARGET_COVERAGE);
+    }
```

Comment:
Can I suggest that you also modify the existing test `ReservoirDownsamplerUnitTest.testReservoirDownsampler()` to check, at the end, that the results from running with a `ReservoirDownsampler` wrapped in a `ReadsDownsamplingIterator` match the results from running standalone?

`PositionalDownsamplerUnitTest.testPositionalDownsampler()` has a check like this at the end, to illustrate what I mean:

```
        // Now test with a PositionalDownsampler wrapped in an iterator, and make sure we get the same results.
        // It's crucial to reset the random number generator again in order to match the selections made by the
        // first downsampling pass.
        Utils.resetRandomGenerator();
        final ReadsDownsamplingIterator downsamplingIter = new ReadsDownsamplingIterator(allReads.iterator(), new PositionalDownsampler(targetCoverage, header));
        final List<GATKRead> downsampledReadsFromIter = new ArrayList<>();
        for ( final GATKRead downsampledRead : downsamplingIter ) {
            downsampledReadsFromIter.add(downsampledRead);
        }

        Assert.assertEquals(downsampledReadsFromIter, downsampledReads, "Results from PositionalDownsampler wrapped in a ReadsDownsamplingIterator do not match results from standalone PositionalDownsampler");
```

Reply:
Yeah good idea. Done.